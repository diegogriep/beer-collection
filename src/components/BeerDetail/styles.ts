import styled from 'styled-components'
import { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    border-radius: ${theme.border.radius};
    padding: ${theme.spacings.small};

    h1 {
      font-size: ${theme.font.sizes.large};
      text-transform: uppercase;
    }

    h2 {
      font-size: ${theme.font.sizes.small};
    }

    h3, h4 {
      padding-top: ${theme.spacings.xsmall};
      text-transform: uppercase;
    }

    article {
      padding: ${theme.spacings.xsmall} 0;
      text-align: center;
    }

    section {
      font-size: ${theme.font.sizes.small};
      padding-top: ${theme.spacings.xsmall};
      text-align: left;

      span {
        display: block;
        padding-top: ${theme.spacings.xsmall};

        strong {
          display: block;
          padding-bottom: ${theme.spacings.xsmall};
          text-transform: uppercase;
        }
      }
    }

    a {
      font-size: ${theme.font.sizes.small};
    }

    ul {
      list-style-type: none;
      margin: 0;
      padding: 0;
    }



    ${media.greaterThan('small')`
      article {
        column-gap: ${theme.spacings.xsmall};
        display: flex;
        text-align: left;
      }
    `}
  `}
`
