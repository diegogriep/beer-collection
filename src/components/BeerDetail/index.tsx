'use client'

import Image from 'next/image'
import { BeerProps, IngredientsKindProps } from '../BeerCard'
import * as S from './styles'
import { handleIngredientObj } from '@/utils/handleIngredientObj'
import { formatIngredient } from '@/utils/formatIngredient'

const BeerDetail = ({
  brewers_tips,
  description,
  image_url,
  ingredients,
  name,
  tagline
}: BeerProps) => {
  const mergedIngredients = handleIngredientObj(ingredients)

  return (
    <S.Wrapper>
      <h1>{name}</h1>
      <h2>{tagline}</h2>

      <article>
        <Image priority alt={name!} src={image_url!} width={180} height={360} />

        <section>
          {description}
          <span>
            <strong>Tips:</strong>
            {brewers_tips}
          </span>

          <h3>Ingredients</h3>
          <ul>
            {mergedIngredients.map(
              (ingredient: string | IngredientsKindProps, i: number) => {
                const topping =
                  typeof ingredient === 'string'
                    ? ingredient
                    : Object.keys(ingredient)

                return (
                  <li key={i}>
                    <h4>{topping}</h4>

                    {typeof ingredient !== 'string' &&
                      formatIngredient(ingredient)}
                  </li>
                )
              }
            )}
          </ul>
        </section>
      </article>
    </S.Wrapper>
  )
}

export default BeerDetail
