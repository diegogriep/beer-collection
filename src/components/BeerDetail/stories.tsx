import { Meta, StoryObj } from '@storybook/react'
import BeerDetail from '.'

import beer from '../BeerCard/mock'
import { BeerProps } from '../BeerCard'

export default {
  title: 'components/BeerDetail',
  component: BeerDetail,
  args: { ...beer[0] }
} as Meta

export const Default: StoryObj<BeerProps> = {
  render: (args) => (
    <BeerDetail {...args} />
  )
}
