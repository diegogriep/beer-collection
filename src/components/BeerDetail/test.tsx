import { screen } from '@testing-library/react'

import BeerDetail from '.'
import { renderWithTheme } from '../../utils/tests/helpers'
import beer from '../BeerCard/mock'

describe('<BeerDetail />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<BeerDetail {...beer[0]} />)

    expect(screen.getByRole('heading', { name: /yeast: Wyeast 1056 - American Aleâ„¢/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
