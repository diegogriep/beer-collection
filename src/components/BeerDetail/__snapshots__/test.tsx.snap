// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BeerDetail /> should render the heading 1`] = `
.c0 {
  background-color: #FAE96F;
  border-radius: 0.4rem;
  color: #000;
  padding: 2.4rem;
}

.c0 h1 {
  font-size: 1.8rem;
  text-transform: uppercase;
}

.c0 h2 {
  font-size: 1.4rem;
}

.c0 h3,
.c0 h4 {
  padding-top: 1.6rem;
  text-transform: uppercase;
}

.c0 article {
  padding: 1.6rem 0;
  text-align: center;
}

.c0 section {
  font-size: 1.4rem;
  padding-top: 1.6rem;
  text-align: left;
}

.c0 section span {
  display: block;
  padding-top: 1.6rem;
}

.c0 section span strong {
  display: block;
  padding-bottom: 1.6rem;
  text-transform: uppercase;
}

.c0 a {
  font-size: 1.4rem;
}

.c0 ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

@media (min-width: 450px) {
  .c0 article {
    align-items: center;
    column-gap: 1.6rem;
    display: flex;
    text-align: left;
  }
}

<main
  class="c0"
>
  <h1>
    Buzz
  </h1>
  <h2>
    A Real Bitter Experience.
  </h2>
  <article>
    <img
      alt="Buzz"
      data-nimg="1"
      decoding="async"
      height="360"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fimages.punkapi.com%2Fv2%2Fkeg.png&w=384&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimages.punkapi.com%2Fv2%2Fkeg.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.punkapi.com%2Fv2%2Fkeg.png&w=384&q=75 2x"
      style="color: transparent;"
      width="180"
    />
    <section>
      A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.
      <span>
        <strong>
          Tips:
        </strong>
        The earthy and floral aromas from the hops can be overpowering. Drop a little Cascade in at the end of the boil to lift the profile with a bit of citrus.
      </span>
      <h3>
        Ingredients
      </h3>
      <ul>
        <li>
          <h4>
            malt
          </h4>
          <p>
            Maris Otter Extra Pale - 3.3 kilograms
          </p>
          <p>
            Caramalt - 0.2 kilograms
          </p>
          <p>
            Munich - 0.4 kilograms
          </p>
        </li>
        <li>
          <h4>
            hops
          </h4>
          <p>
            Fuggles - 25 grams
          </p>
          <p>
            First Gold - 25 grams
          </p>
          <p>
            Fuggles - 37.5 grams
          </p>
          <p>
            First Gold - 37.5 grams
          </p>
          <p>
            Cascade - 37.5 grams
          </p>
        </li>
        <li>
          <h4>
            yeast: Wyeast 1056 - American Aleâ„¢
          </h4>
        </li>
      </ul>
    </section>
  </article>
</main>
`;
